src/transformers/pipelines/base.py

```diff
@@ -149,7 +149,7 @@ def inner(items):
                 _padding_value = t_padding_value
             elif key in {"input_values", "pixel_values", "input_features"}:
                 _padding_value = f_padding_value
-            elif key in {"p_mask"}:
+            elif key in {"p_mask", "special_tokens_mask"}:
```

Comment:
In random models, `special_tokens_mask` would be extended in the batch with `0` instead of `1` so we could still predict `PAD` token in the pipeline.

I think having `pad` being always considered a `special_tokens_mask` is fine.